### PR TITLE
Features/#384 inspections are enabled for workflows

### PR DESCRIPTION
**Pull request by dakochik**
Issue #384 is resolved

### CODE REVIEW

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnexpectedKeywordArgsInspection.kt [hunk 1]
```diff
@@ -27,6 +28,12 @@ class SmkSectionUnexpectedKeywordArgsInspection : SnakemakeInspection() {
             }
         }
 
+        override fun visitSmkWorkflowArgsSection(st: SmkWorkflowArgsSection) {
+            if (st.sectionKeyword in SECTIONS_WHERE_KEYWORD_ARGS_PROHIBITED) {
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnexpectedKeywordArgsInspection.kt:32]: You check workflow top-level section name in list of rule/checkpoint section names. Here should be 2 different lists, one for rule/checkpoint set, the other for workflow top level sections. E.g. for 'container' we have SnakemakeNames.WORKFLOW_CONTAINER_KEYWORD and SnakemakeNames.CONTAINER_KEYWORD.

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionMultipleArgsInspection.kt [hunk 2]
```diff
@@ -25,6 +26,13 @@ class SmkSectionMultipleArgsInspection : SnakemakeInspection() {
             }
         }
 
+        override fun visitSmkWorkflowArgsSection(st: SmkWorkflowArgsSection) {
+            val name = st.sectionKeyword
+            if (name != null && name in SINGLE_ARGUMENT_SECTIONS_KEYWORDS) {
```
Comment (iromeo) [inline src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionMultipleArgsInspection.kt:31]: Same comment as in SmkSectionUnexpectedKeywordArgsInspection

**Review by iromeo — CHANGES_REQUESTED** [review]
Workflow and rule/checkpoint names should be decoupled + please fix also `SmkSyntaxErrorAnnotator` to highlight as error:
```
wildcard_constraints:
    foo = ".*",
    foo = ".*"
```

**Comment by dakochik** [comment]
fixed: highlighting for SmkSyntaxErrorAnnotator
refactored: separate sections for workflows keywords are created

**Pull request merged**